File: day1/1.jl
f = open("./day1/input.txt")
numbers = [parse(Int32, i) for i in readlines(f)]
close(f)



function solve(target = 2020) 
    diffs = Dict()

    for num in numbers

        diff = target - num

        if num in keys(diffs)
            return num*diff
        end
        diffs[diff] = num
    end
            
end

print(solve(), "\n")

File: day1/1.py
with open("input.txt") as f:
    numbers = [int(num) for num in f.readlines()]

def solve(target = 2020):

    diffs = {}
    for num in numbers:
        diff = target - num

        if num in diffs:
            return num*diff

        diffs[diff] = num

if __name__ == "__main__":
    print(solve())

